### PR TITLE
[libc++][modules] Uses _LIBCPP_USING_IF_EXISTS.

### PR DESCRIPTION
**Pull request by mordante**
This attribute is used in the headers. Not using this in the modules has led to several issues. Add them to the modules to avoid these errors in other placed.

### CODE REVIEW

File: libcxx/modules/std/atomic.inc [hunk 1]
```diff
@@ -10,132 +10,132 @@
...
 
   // [atomics.order], order and consistency
-  using std::memory_order;
-  using std::memory_order_acq_rel;
-  using std::memory_order_acquire;
-  using std::memory_order_consume;
-  using std::memory_order_relaxed;
-  using std::memory_order_release;
-  using std::memory_order_seq_cst;
+  using std::memory_order _LIBCPP_USING_IF_EXISTS;
```
Comment (philnik777) [inline libcxx/modules/std/atomic.inc:13]: We provide (almost?) all of the symbols listed here unconditionally. Why are they marked `_LIBCPP_USING_IF_EXISTS`? These are also not marked in the `std.compat` module.

Comment (ldionne) [reply libcxx/modules/std/atomic.inc:13]: I have no idea why we mark those with `_LIBCPP_USING_IF_EXISTS` in the headers. We shouldn't. However I suggest we blindly apply this patch to match the headers, and then fix the remaining issues in both the headers and the module file.

File: libcxx/modules/std.compat/cmath.inc [hunk 2]
```diff
@@ -8,241 +8,241 @@
...
 export {
-  using ::double_t;
-  using ::float_t;
+  using ::double_t _LIBCPP_USING_IF_EXISTS;
+  using ::float_t _LIBCPP_USING_IF_EXISTS;
 
-  using ::acos;
-  using ::acosf;
-  using ::acosl;
+  using ::acos _LIBCPP_USING_IF_EXISTS;
```
[the remark this answers is not in the document]
Comment (mordante) [reply libcxx/modules/std.compat/cmath.inc:14]: Are we sure? I can imagine embedded platforms without floating-point support may not have these.

File: libcxx/modules/std/atomic.inc (hunk 1 above)
Comment (mordante) [reply libcxx/modules/std/atomic.inc:13]: These values are from C headers too.
I indeed added what we did for the C-headers. Before I did not add all of these and that lead to several issues. So I agree with @ldionne's approach.

File: libcxx/modules/std.compat/cmath.inc (hunk 2 above)
[the remark this answers is not in the document]
Comment (philnik777) [reply libcxx/modules/std.compat/cmath.inc:14]: Yes. We provide them even if the libc doesn't. It becomes a linker error if it's used. See `__math/*` for the definitions.

File: libcxx/modules/std.compat/cstdlib.inc [hunk 3]
```diff
@@ -8,65 +8,65 @@
...
-  using ::at_quick_exit _LIBCPP_USING_IF_EXISTS;
-  using ::atexit;
-  using ::exit;
-  using ::quick_exit _LIBCPP_USING_IF_EXISTS;
+  using ::_Exit _LIBCPP_USING_IF_EXISTS;
+  using ::abort _LIBCPP_USING_IF_EXISTS;
+  using ::at_quick_exit _LIBCPP_USING_IF_EXISTS _LIBCPP_USING_IF_EXISTS;
+  using ::atexit _LIBCPP_USING_IF_EXISTS;
+  using ::exit _LIBCPP_USING_IF_EXISTS;
+  using ::quick_exit _LIBCPP_USING_IF_EXISTS _LIBCPP_USING_IF_EXISTS;
```
Comment (ldionne) [inline libcxx/modules/std.compat/cstdlib.inc:22]: There's a few cases where this was added twice by mistake.

**Review by ldionne — APPROVED** [review]
LGTM w/ duplicate attributes fixed.

**Comment by EricWF** [comment]
> This attribute is used in the headers. Not using this in the modules has led to several issues. Add them to the modules to avoid these errors in other placed.

If  they're not needed without modules, then I suspect this is just papering over the root cause of the breakages.
Could you point me towards some concrete examples?

**Comment by mordante** [comment]
> > This attribute is used in the headers. Not using this in the modules has led to several issues. Add them to the modules to avoid these errors in other placed.
> 
> If they're not needed without modules, then I suspect this is just papering over the root cause of the breakages. Could you point me towards some concrete examples?

We've discussed this privately.

**Pull request merged**